examples/TEM_data/README.md

```diff
@@ -0,0 +1,381 @@
+  as input and return one or more data models as output. It is implemented
+  as a Python function.  Conversions are similar to property mappings,
+  but operates on instances instead of individual properties.
+  The *TEM image analyser* is implemented as a converter.  It takes
+  an instance of a [TEMImage] data model and returns an instance of
+  a [PrecipitateStatistics] data model.
+
+
+Setup
+-----
+Create a new virtual environment and install needed packages
+
+    pip install -r requirements.txt
+
+
+Running the example
+-------------------
+The main example can be found in the [demo.py] script. We will here go
+through it step by step.
+
+### Initialisation
+In addition to define the input and output data directories to use in this
+example, the line
+
+```python
+from paths import indir, outdir
+```
+
+also sets up search paths for DLite and include this example directory in
+the Python module search path (see [paths.py]):
+
+```python
+dlite.storage_path.append(entitydir)
+dlite.python_storage_plugin_path.append(plugindir)
+
+sys.path.append(str(exampledir))
+```
+
+The line
+
+```python
+from utils.utils import get_data, populate_triplestore
+```
+
+imports the two main functions that we will use in this example.
+
+
+### Accessing the knowledge base
+To be independent of the underlying triplestore implementation, we use
+tripper to access the knowledge base.  For the purpose of this demo we
+use the *rdflib* backend.  We also prepare the knowledge base by
+populating it with the documentation of all the data resources from
+the YAML file [data/resources.yaml].
+
+```python
+ts = Triplestore(backend="rdflib")
+populate_triplestore(ts, indir / "resources.yaml")
+```
+
+Lets also define some handy tripper namespaces
+
+```python
+OTEIO = Namespace("http://emmo.info/oteio#")
+PM = ts.bind("pm", "https://www.ntnu.edu/physmet/data#")
+```
+
+
+### Listing the data resources in the knowledge base
+Lets see the data sources and sinks that we have documented.
+
+```python
+# List all data sources and sinks
+>>> print("Data sources:")
+>>> for source in ts.subjects(RDF.type, OTEIO.DataSource):
+...     print("  -", source)
+>>> print()
+>>> print("Data sinks:")
+>>> for sink in ts.subjects(RDF.type, OTEIO.DataSink):
+...     print("  -", sink)
+Data sources:
+  - https://www.ntnu.edu/physmet/data#TEM_BF_100-at-m5-and-2_001
+  - https://www.ntnu.edu/physmet/data#TEM_040
+  - https://www.ntnu.edu/physmet/data#TEM_6c8cm_008
+  - https://www.ntnu.edu/physmet/data#alloy_composition
+
+Data sinks:
+  - https://www.ntnu.edu/physmet/data#thumbnail_image
+  - https://www.ntnu.edu/physmet/data#microscope_settings
+  - https://www.ntnu.edu/physmet/data#image_analyser
+  - https://www.ntnu.edu/physmet/data#precipitate_statistics
+  - https://www.ntnu.edu/physmet/data#precipitation_model_input
+```
+
+(See the OntoTrans [Exploratory Search System (ESS)] by TU Wien
+for an example of how to data in a knowledge base can be explored.)
```

Comment:
```suggestion
for an example of how to put data in a knowledge base can be explored.)
```
If we want to have this in dlite we should have a link.

Reply:
ESS only explores what already is in the knowledge base.

Not sure if I understand the second comment?

Reply:
the "put" is missing. I see that the link to the ESS is there, but it is not working for me.